cli/dts/lib.deno.unstable.d.ts

```diff
@@ -516,6 +522,40 @@ declare namespace Deno {
     options?: EmitOptions,
   ): Promise<EmitResult>;
 
+  /**
+   * **UNSTABLE**: new API, yet to be vetted.
+   *
+   * The options for `Deno.emitBundle()` API.
+   */
+  export interface EmitBundleOptions extends EmitOptions {
+    /** Indicate that the source code should be emitted to a single file
+     * JavaScript bundle that is a single ES module (default: `"module"`) or a
+     * single file self contained script we executes in an immediately invoked
+     * function when loaded (`"classic"`). */
+    type?: "module" | "classic";
+  }
+
+  /**
+   * **UNSTABLE**: new API, yet to be vetted.
+   *
+   * The result of `Deno.emitBundle()` API.
+   */
+  export type EmitBundleResult = Omit<EmitResult, "modules"> & {
```

Comment:
I really dislike these complex types. They are very ugly, and are not rendered nicely on the docs.